src/solver/impls/petsc/petsc.cxx

```diff
@@ -359,26 +365,13 @@ int PetscSolver::init(int NOUT, BoutReal TIMESTEP) {
     ierr = PetscViewerDestroy(&fd);CHKERRQ(ierr);
   } else { // create Jacobian matrix
 
-    //PetscInt MXSUB = mesh->xend - mesh->xstart + 1;
-    //PetscInt MYSUB = mesh->yend - mesh->ystart + 1;
-
-    PetscInt nx = mesh->xend;//MXSUB;
-    PetscInt ny = mesh->yend;//MYSUB;
-
-    /* number of z points */
-    PetscInt nz  = mesh->LocalNz;
-
     /* number of degrees (variables) at each grid point */
     PetscInt dof = n3Dvars();
```

Comment:
I think this code might need generalising to handle 2Ds as well

Reply:
I think it probably does already handle 2Ds ok. This bit is for the preconditioner, and only if the user hasn't supplied their own. And then it's only for the preallocation of J and is just an optimisation.

Certainly it works with a Field2D in the test-solver